enhancements/installer/alibaba-cloud-ipi.md

```diff
@@ -0,0 +1,195 @@
+4. Add a CI job that executes the provisioning steps to test the Alibaba IPI installer.
+
+### Non-Goals
+
+## Proposal
+
+### User Stories
+
+Enables `openshift-install` to create and destroy a cluster on Alibaba Cloud, which is managed and released and covered by CI tests.
+
+### Implementation Details/Notes/Constraints [optional]
+
+1. The default installation is going to have below Alibaba resources.
+- ControlPlane:
+  - name: master
+  - availability zones: 3
+  - size: 500
+  - PerformanceLevel: PL1
+  - Category: ESSD
+  - InstanceType: g6a.xlarge
+
+- Compute:
+  - name: worker
+  - availability zones: 3
+  - size: 500
+  - PerformanceLevel: PL1
+  - Category: ESSD
+  - InstanceType: c6.4xlarge
+
+- Temporary:
+  - name: bootstrap
+  - availability zones: 1
+  - size: 100
+  - PerformanceLevel: PL1
+  - Category: ESSD
+  - InstanceType: g6a.xlarge
+
+- SLB:
+  - Option One:
+    - 1 for external access
+    - 1 for internal access
+  - Option Two:
+    - 1 for external and internal access. external need to bind to EIP.
+
+- Public IP:
+  - 3 EIP's for NATGateway
+  - 1 EIP binding to SLB if Option Two is chosen.
+
+- DNS:
+  - 1 Alibaba Cloud DNS - privatezone for private domain name resolution.
+  - 1 Alibaba Cooud DNS - Manage DNS for public domain name resolution.
+
+2. We're planning to use Alibaba Cloud DNS and privatezone solution. The nodes within the VPC utilize [privatezone for private domain name resolution](https://www.alibabacloud.com/help/product/64583.html) to provide internal DNS feature. [Manage DNS for public domain name resolution](https://www.alibabacloud.com/product/dns) will provide public DNS feature.
+
+3. We're going to use one public-facing and one private [load balancer](https://www.alibabacloud.com/help/doc-detail/196874.htm) for internal/external requests to the API. Option One is one for external access and one for internal access. Option Two is one for both extrenal and internal access. but external need to bind one EIP.
+Now it is going to create load balancing based on nginx ingress controller. Also, service type load balancing can use [Alibaba CLB](https://www.alibabacloud.com/help/doc-detail/27539.htm).
+
+4. We are planing to use tags withadditional keys and values for installer to all resources that it creates.
+
+5. We're going to use credentials. Long-lived tokens uses [Alibaba RAM AK](https://www.alibabacloud.com/help/doc-detail/122151.htm).
+Short-lived tokens uses [Alibaba Cloud Security Token Service (STS)](https://www.alibabacloud.com/help/doc-detail/28756.html)
+
+### Risks and Mitigations
+
+If users expect differences in the resources and their available options across different regions. users can expect the same level of performance. But some resource available in most regions might be uavailable in some region. The solution is that we could provide preferred regions list to make sure the resources could be available on list.  
```

Comment:
What sorts of resources might not be available in some regions?

Reply:
Region differences:
- Number of availability zones
- storage differences 

Default instance types should be the same for each region.